src/main/java/com/palantir/docker/compose/configuration/EnvironmentValidator.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.docker.compose.configuration;
+
+import java.util.Map;
+
+public interface EnvironmentValidator {
+
+    Map<String, String> validate(Map<String, String> dockerEnvironment);
```

Comment:
As someone who hasn't encountered this part of the code before, it wasn't immediately obvious to me what this interface is for.

Perhaps we could use some more descriptive types, method names or javadoc to make this super obvious?  Also I couldn't see why this method actually returns another Map<String,String>... What about:

```java
void validateEnvironmentVariables(Map<String, String> environmentVariables);
```